automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/Orchestration.scala

```diff
@@ -0,0 +1,185 @@
+    op() match {
+      case true => true
+      case false if remainingBackOffIntervals.isEmpty => false
+      case false if remainingBackOffIntervals.nonEmpty =>
+        Thread sleep remainingBackOffIntervals.head.toMillis
+        retry(remainingBackOffIntervals.tail)(op)
+    }
+  }
+
+  object billing {
+
+    object BillingProjectRole extends Enumeration {
+      val User = Value("user")
+      val Owner = Value("owner")
+    }
+
+    def addUserToBillingProject(projectName: String, email: String, role: BillingProjectRole.Value)(implicit token: AuthToken): Unit = {
+      putRequest(Config.FireCloud.orchApiUrl + s"api/billing/$projectName/${role.toString}/$email")
+    }
+
+    def createBillingProject(projectName: String, billingAccount: String)(implicit token: AuthToken): Unit = {
+      postRequest(Config.FireCloud.orchApiUrl + "api/billing", Map("projectName" -> projectName, "billingAccount" -> billingAccount))
+
+      retry(Seq.fill(30)(10.seconds)) { () =>
+        val response: String = parseResponse(getRequest(Config.FireCloud.orchApiUrl + "api/profile/billing"))
+        val projects: List[Map[String, Object]] = responseAsList(response)
+        projects.exists((e) =>
+          e.exists(_ == ("creationStatus", "Ready")) && e.exists(_ == ("projectName", projectName)))
+      } match {
+        case false => throw new Exception("Billing project creation did not complete")
+        case true => println("Finished creating billing project")
+      }
+    }
+  }
+
+  object groups {
+
+    object GroupRole extends Enumeration {
+      val Member = Value("member")
+      val Admin = Value("admin")
+    }
+
+    def addUserToGroup(groupName: String, email: String, role: GroupRole.Value)(implicit token: AuthToken): Unit = {
+      logger.info(s"Adding user to group: $groupName $email")
+      putRequest(Config.FireCloud.orchApiUrl + s"api/groups/$groupName/${role.toString}/$email")
+    }
+
+    def create(groupName: String)(implicit token: AuthToken): Unit = {
+      logger.info(s"Creating group: $groupName")
+      postRequest(Config.FireCloud.orchApiUrl + s"api/groups/$groupName")
+    }
+
+    def delete(groupName: String)(implicit token: AuthToken): Unit = {
+      logger.info(s"Deleting group: $groupName")
+      deleteRequest(Config.FireCloud.orchApiUrl + s"api/groups/$groupName")
+    }
+
+    def removeUserFromGroup(groupName: String, email: String, role: GroupRole.Value)(implicit token: AuthToken): Unit = {
+      logger.info(s"Removing user from group: $groupName $email")
+      deleteRequest(Config.FireCloud.orchApiUrl + s"api/groups/$groupName/${role.toString}/$email")
+    }
+  }
+
+  object workspaces {
+
+    def create(namespace: String, name: String, authDomain: String = "")
+              (implicit token: AuthToken): Unit = {
+      logger.info(s"Creating workspace: $namespace/$name with auth domain: $authDomain")
+
+      val authDomainMap = authDomain match {
```

Comment:
I had that originally but didn't like how it looked when calling. Either the argument would be omitted or it would always be `Option["blah"]` which seemed redundant to me. The code is currently split between the 2 options but I definitely want to consistently use one or the other. Is there a strong argument for using an `Option`?